plugins/dashboard_plugin.py

```diff
@@ -96,53 +67,69 @@ def list(self):
             'bills_in_db': bills_in_db,
             'bills_in_index': bills_in_index,
+            'datetime_format': self.DATETIME_FORMAT,
         }
 
         return self.render_template('dashboard.html', **metadata)
 
-    def get_dag_info(self, dags, session):
+    def get_dag_info(self):
+
+        dags = [self.AIRFLOW_DAG_BAG.get_dag(dag_id) for dag_id in self.AIRFLOW_DAG_BAG.dag_ids
+                if not dag_id.startswith('airflow_')]  # Filter meta-DAGs
+
         data = []
+
         for d in dags:
-            last_run = dag.get_last_dagrun(d.dag_id, session,include_externally_triggered=True)
+            last_run = dag.get_last_dagrun(d.dag_id,
+                                           self.AIRFLOW_SESSION,
+                                           include_externally_triggered=True)
 
             if last_run:
                 run_state = last_run.get_state()
+                run_date_info = self._get_localized_time(last_run.execution_date)
 
-                run_date = last_run.execution_date
+                last_successful, last_successful_info = self._get_last_successful_dagrun(d)
 
-                pst_run_time = run_date.astimezone(PACIFIC_TIMEZONE)
-                cst_run_time = run_date.astimezone(CENTRAL_TIMEZONE)
-                run_date_info = {
-                    'pst_time': datetime.strftime(pst_run_time, "%m/%d/%y %I:%M %p"),
-                    'cst_time': datetime.strftime(cst_run_time, "%m/%d/%y %I:%M %p")
-                }
+                next_scheduled = d.following_schedule(datetime.now(pytz.utc))
+                next_scheduled_info = self._get_localized_time(next_scheduled)
 
-                now = datetime.now(pytz.utc)
-                next_scheduled = d.following_schedule(now)
-
-                pst_next_scheduled = next_scheduled.astimezone(PACIFIC_TIMEZONE)
-                cst_next_scheduled = next_scheduled.astimezone(CENTRAL_TIMEZONE)
-                next_scheduled_info = {
-                    'pst_time': datetime.strftime(pst_next_scheduled, "%m/%d/%y %I:%M %p"),
-                    'cst_time': datetime.strftime(cst_next_scheduled, "%m/%d/%y %I:%M %p")
-                }
             else:
                 run_state = None
-                run_date_info = None
-                next_scheduled_info = None
+                run_date_info = {}
+
+                last_successful = None
+                last_successful_info = {}
+
+                next_scheduled_info = {}
 
             dag_info = {
                 'name': d.dag_id,
                 'description': d.description,
                 'run_state': run_state,
                 'run_date': run_date_info,
+                'last_successful': last_successful,
+                'last_successful_date': last_successful_info,
                 'next_scheduled': next_scheduled_info
             }
 
             data.append(dag_info)
 
         return data
 
+    def get_last_successful_dagrun(self, dags):
```

Comment:
It's a little confusing to me that we have a public method and a private method with basically the same name (`get_last_successful_dagrun`) but that do two different things. Can we adjust the name of one of these methods to make it clearer?

Reply:
Sure thing!